clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java

```diff
@@ -410,6 +330,37 @@ private FilesetContextPair getFilesetContext(Path virtualPath, FilesetDataOperat
     return new FilesetContextPair(new Path(actualFileLocation), fs);
   }
 
+  private Map<String, String> getCredentialProperties(
+      FileSystemProvider fileSystemProvider,
+      Catalog catalog,
+      NameIdentifier filesetIdentifier,
+      String virtualPathString) {
+    Map<String, String> maps = Maps.newHashMap();
+    try {
+      Fileset fileset =
+          catalog
+              .asFilesetCatalog()
+              .loadFileset(
+                  NameIdentifier.of(
+                      filesetIdentifier.namespace().level(2), filesetIdentifier.name()));
+      Credential[] credentials = fileset.supportsCredentials().getCredentials();
+      if (credentials.length > 0 && fileSystemProvider instanceof SupportsCredentialVending) {
```

Comment:
could you get credentials when `fileSystemProvider` is `SupportsCredentialVending`?